src/olympia/devhub/forms.py

```diff
@@ -1062,31 +1062,40 @@ def check_blocklist(self, guid, version_string):
             raise forms.ValidationError(formatted_msg)
 
     def check_for_existing_versions(self, version_string):
-        if self.addon:
-            # Make sure we don't already have this version.
-            existing_versions = Version.unfiltered.filter(
-                addon=self.addon, version=version_string
-            )
-            if existing_versions.exists():
-                version = existing_versions[0]
-                if version.deleted:
-                    msg = gettext('Version {version} was uploaded before and deleted.')
-                elif version.file.status == amo.STATUS_AWAITING_REVIEW:
-                    next_url = reverse(
-                        'devhub.submit.version.details',
-                        args=[self.addon.slug, version.pk],
-                    )
-                    msg = DoubleSafe(
-                        '%s <a href="%s">%s</a>'
-                        % (
-                            gettext('Version {version} already exists.'),
-                            next_url,
-                            gettext('Continue with existing upload instead?'),
-                        )
+        # Make sure we don't already have this version.
+        existing_versions = Version.unfiltered.filter(
+            addon=self.addon, version=version_string
+        )
+        if existing_versions.exists():
+            version = existing_versions[0]
+            if version.deleted:
+                msg = gettext('Version {version} was uploaded before and deleted.')
+            elif version.file.status == amo.STATUS_AWAITING_REVIEW:
+                next_url = reverse(
+                    'devhub.submit.version.details',
+                    args=[self.addon.slug, version.pk],
+                )
+                msg = DoubleSafe(
+                    '%s <a href="%s">%s</a>'
+                    % (
+                        gettext('Version {version} already exists.'),
+                        next_url,
+                        gettext('Continue with existing upload instead?'),
                     )
-                else:
-                    msg = gettext('Version {version} already exists.')
-                raise forms.ValidationError(msg.format(version=version_string))
+                )
+            else:
+                msg = gettext('Version {version} already exists.')
+            raise forms.ValidationError(msg.format(version=version))
+
+    def check_for_greater_version_number(self, version_string):
+        if (
```

Comment:
I know it's just a couple of lines of code, but this is the same check that was added into `src/olympia/addons/serializers.py`, so should we maybe have a shared helper for the logic?

Reply:
I thought the same when I was writing it.  The logic is literally just the one line: `previous_version.version >= version_string` (and then we throw a slightly different exception in each place) so I wasn't sure it would make a difference.  I'll try a helper function and see how that looks.

Reply:
Yeah, it just being a single line of code it does seem like it could be overkill, but because it's business logic it's nice to have it in once place if for some reason in the future the logic needs to change.

Reply:
https://github.com/mozilla/addons-server/pull/19820/commits/d29edd74bf304c0163fd4fbc1ca7125b0d6e2997  not bad